### PR TITLE
Generate a default admin user

### PR DESCRIPTION
**Pull request by hardillb**
part of #2436

## Description

This will create a new admin user when the setup wizard runs if none already exist and `create_admin: true` in the `etc/flowforge.yml` file

This is to meet a AWS requirement

## Related Issue(s)

#2436 

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/flowforge/flowforge/blob/main/CONTRIBUTING.md)
 - [x] Suitable unit/system level tests have been added and they pass 
 - [x] Documentation has been updated
    - [ ] Upgrade instructions
    - [x] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `flowforge/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `flowforge/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Backport needed? -> add the `backport` label
 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

**Comment by hardillb** [comment]
Not sure where to test this at the moment.

**Comment by hardillb** [comment]
I've just noticed that `admin` is banned as a username in the setup wizard, so what should we change this to?

**Comment by knolleary** [comment]
Maybe `ff-admin` ?

File: forge/routes/setup/index.js [hunk 1]
```diff
@@ -19,6 +25,25 @@ module.exports = async function (app) {
             reply.redirect('/')
             return
         }
+
+        if (app.config.create_admin) {
+            if (await app.db.models.User.count() === 0) {
+                const password = generatePassword()
+                await app.db.models.User.create({
+                    username: 'admin',
```
Comment (hardillb) [inline forge/routes/setup/index.js:33]:
```suggestion
                    username: 'ff-admin',
```

File: forge/routes/setup/index.js [hunk 2]
```diff
@@ -19,6 +25,25 @@ module.exports = async function (app) {
...
+                    name: 'Default Admin',
+                    email: 'admin@example.com',
+                    email_verified: true,
+                    password,
+                    admin: true,
+                    password_expired: true
+                })
+                app.log.info('[SETUP] Created default Admin User')
+                app.log.info('[SETUP] username: admin')
```
Comment (hardillb) [inline forge/routes/setup/index.js:42]:
```suggestion
                app.log.info('[SETUP] username: ff-admin')
```

File: test/system/002-setup-admin_spec.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        cookies[0].should.have.property('name', '_csrf')
+        CSRF_COOKIE[cookies[0].name] = cookies[0].value
+
+        const admin = await forge.db.models.User.findOne({
+            where: {
+                admin: true
+            }
+        })
+
+        admin.should.have.property('username', 'admin')
```
Comment (hardillb) [inline test/system/002-setup-admin_spec.js:58]:
```suggestion
        admin.should.have.property('username', 'ff-admin')
```

**Comment by hardillb** [comment]
Updated

File: docs/install/configuration.md [hunk 4]
```diff
@@ -20,6 +20,7 @@ Option | Description
 `base_url` | The url to access the platform. This defaults to `http://localhost:3000` which means a number of internally generated URLs will only work when browsing on the same device as is running the platform. To be able to access the platform remotely, replace `localhost` with the ip address of the device running FlowForge. 
 `domain` | The domain that instance names will be pre-pended to on Docker & Kubernetes platforms to create a hostname to access the instance. A wildcard DNS A record should point be configured to point to the FlowForge entry IP Address.
 `support_contact` | a URL or string with contact details for the administrator e.g `mailto:support@example.com` or `https://support.example.com` . Defaults to the email address of the first admin user or `the administrator` if no email address set.
+`create_admin` | If set to `true` will create a default admin user on first run, the username/password is written to the logs .Default: `false`
```
Comment (hardillb) [inline docs/install/configuration.md:23]:
```suggestion
`create_admin` | If set to `true` will create a default admin user on first run, the username/password is written to the logs. Default: `false`
```

File: etc/flowforge.yml [hunk 5]
```diff
@@ -95,3 +95,8 @@ email:
 #################################################
 telemetry:
   enabled: true
+
+#################################################
+# Create Default Admin                          #
+#################################################
+# create_admin: false
```
Comment (hardillb) [inline etc/flowforge.yml:102]:
```suggestion
# create_admin: false

```

**Comment by hardillb** [comment]
Will merge when tests finish

**Pull request merged**